src/schain.js

```diff
@@ -0,0 +1,112 @@
+import { isNumber, isStr, isArray, isPlainObject, hasOwn, logError, isInteger } from './utils';
```

Comment:
should this be part of core or a separate module?

Reply:
Agree @mkendall07 , i will move schain as a separate module as all pubs may not need it.

Reply:
Moved the code as a module. Resolving the conversation